packages/playwright-test/src/runner.ts

```diff
@@ -448,7 +448,26 @@ export class Runner {
       let sigintWatcher;
 
       let hasWorkerErrors = false;
-      for (const testGroups of concurrentTestGroups) {
+      let previousStageFailed = false;
+      for (let testGroups of concurrentTestGroups) {
+        if (previousStageFailed) {
+          const runAlwaysGroups = [];
```

Comment:
Let's extract a method `skipTestsNotMarkedAsRunAlways()` or something like that?